src/ja/worker/proxy/proxy.py

```diff
@@ -0,0 +1,42 @@
+from ja.common.proxy.proxy import ContinuousProxy
+from ja.common.message.server import ServerResponse
+from ja.common.proxy.ssh import SSHConfig
+
+
+class WorkerServerProxy(ContinuousProxy):
+    """
+    Proxy for the central server used on the worker client.
+    """
+    def __init__(self, ssh_config: SSHConfig):
+        """!
+        @param ssh_config: Config for paramiko
+        """
+
+    def register_self(self) -> ServerResponse:
```

Comment:
This should also have a parameter indicating the resources of the work machine.

Reply:
Added as TODO for now, I'll fix it when I rebase.